executor/new_distsql.go

```diff
@@ -110,6 +110,18 @@ func (e *TableReaderExecutor) doRequest() error {
 	return nil
 }
 
+// doRequestForHandles constructs kv ranges by handles. It is used by index look up executor.
+func (e *TableReaderExecutor) doRequestForHandles(handles []int64) error {
+	kvRanges := tableHandlesToKVRanges(e.tableID, handles)
+	var err error
+	e.result, err = distsql.SelectDAG(e.ctx.GetClient(), goctx.Background(), e.dagPB, kvRanges, e.ctx.GetSessionVars().DistSQLScanConcurrency, e.keepOrder, e.desc)
```

Comment:
Use `e.ctx.GoCtx()` instead of `goctx.Backgroud()` makes the request cacelable.
But this introduces lock contention.

The best way is to create a new context by `WithCancel` for each task worker.